### PR TITLE
feat: Add real tests for splunk logger using vector

### PR DESCRIPTION
**Pull request by Revolyssup**
### Description
Fixes https://github.com/apache/apisix/issues/9489
This PR replaces mock splunk-hec-logger tests with real end to end test using vector as log server.

### Checklist

- [x] I have explained the need for this PR and the problem it solves
- [x] I have explained the changes or the new features added to this PR
- [x] I have added tests corresponding to this change
- [ ] I have updated the documentation to reflect this change
- [ ] I have verified that this change is backward compatible (If not, please discuss on the [APISIX mailing list](https://github.com/apache/apisix/tree/master#community) first)

### CODE REVIEW

**Comment by monkeyDluffy6017** [comment]
@Sn0rt please help to review

File: t/plugin/splunk-hec-logging.t [hunk 1]
```diff
@@ -406,19 +392,23 @@ the mock backend is hit
...
-["GET /hello", "GET /hello", "GET /hello"]
---- wait: 2
+=== TEST 12: check splunk log
+--- exec
+tail -n 1 ci/pod/vector/splunk.log
 --- response_body eval
-["hello world\n", "hello world\n", "hello world\n"]
---- no_error_log
-[error]
+qr/.*test batched data.*/
```
Comment (Sn0rt) [inline t/plugin/splunk-hec-logging.t:414]: new line

File: ci/pod/docker-compose.plugin.yml [hunk 2]
```diff
@@ -187,18 +187,6 @@ services:
     networks:
       opa_net:
 
-  # Splunk HEC Logging Service
-  splunk:
```
Comment (Sn0rt) [inline ci/pod/docker-compose.plugin.yml:191]: why remove this component?

Comment (Revolyssup) [reply ci/pod/docker-compose.plugin.yml:191]: Because we don't need a separate splunk server now. Vector does the job

Comment (Sn0rt) [reply ci/pod/docker-compose.plugin.yml:191]: ok

**Comment by monkeyDluffy6017** [comment]
please resolve conflicts

File: t/plugin/splunk-hec-logging.t [hunk 3]
```diff
@@ -406,19 +400,23 @@ the mock backend is hit
             if code >= 300 then
                 ngx.status = code
             end
-            ngx.say(body)
+            local code, _, body2 = t("/hello", "GET")
```
Comment (monkeyDluffy6017) [inline t/plugin/splunk-hec-logging.t:403]: need a blank

File: t/plugin/splunk-hec-logging.t [hunk 4]
```diff
@@ -383,12 +361,28 @@ the mock backend is hit
...
+                            "vip": "$remote_addr",
+                            "message_3":"test batched data"
+                        }
+                }]]
+                )
+            if code >= 300 then
+                ngx.status = code
+                ngx.say(body)
+                return
+            end
```
Comment (monkeyDluffy6017) [inline t/plugin/splunk-hec-logging.t:379]: need a blank

File: t/plugin/splunk-hec-logging.t [hunk 5]
```diff
@@ -224,157 +224,135 @@ hello world
...
-            local code, body = t('/apisix/admin/routes/1', ngx.HTTP_PUT, config)
-
+            })
             if code >= 300 then
                 ngx.status = code
                 ngx.say(body)
                 return
             end
-
+            local code, _, body2 = t("/hello", "GET")
```
Comment (monkeyDluffy6017) [inline t/plugin/splunk-hec-logging.t:336]: need a blank

File: t/plugin/splunk-hec-logging.t [hunk 6]
```diff
@@ -224,157 +224,135 @@ hello world
...
+                            "message_2":"logger format in plugin"
+                        }
+                }]]
+                )
+            if code >= 300 then
+                ngx.status = code
+                ngx.say(body)
+                return
+            end
+            local code, body = t('/apisix/admin/routes/1', ngx.HTTP_PUT, {
```
Comment (monkeyDluffy6017) [inline t/plugin/splunk-hec-logging.t:312]: need a blank

Comment (Revolyssup) [reply t/plugin/splunk-hec-logging.t:313]: Added newline at all suggested place @monkeyDluffy6017

**Comment by monkeyDluffy6017** [comment]
Could you merge the master? The ci always failed because the go packages faile to download.

**Comment by Revolyssup** [comment]
> Could you merge the master? The ci always failed because the go packages faile to download.

@monkeyDluffy6017 I merged master. Please review

**Review by monkeyDluffy6017 — APPROVED** [review]

**Review by leslie-tsang — APPROVED** [review]

**Pull request merged**